exdb/views.py

```diff
@@ -368,6 +368,97 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class CompletionBoardView(TemplateView):
+    access_level = 'basic'
+    template_name = 'exdb/completion_board.html'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(CompletionBoardView, self).get_context_data(*args, **kwargs)
+
+        if not self.request.user.is_hallstaff():
+            raise Http404('User does not have access to the global completion board')
+
+        affiliation_pk = self.kwargs.get('pk') or self.request.user.affiliation_id
+        if affiliation_pk is not None:
+            affiliation = get_object_or_404(Affiliation, pk=affiliation_pk)
+        else:
+            # Just get the first one for now
+            affiliation = Affiliation.objects.first()
+        if affiliation is None:
+            raise Http404('No Affiliation objects found')
+
+        semester = Semester.get_current()
+        if semester is None:
+            raise Http404('No Semester objects found')
+
+        prefetch = Prefetch(
+            'experience_set',
+            queryset=Experience.objects.filter(
+                start_datetime__lte=semester.end_datetime,
+                end_datetime__gte=semester.start_datetime,
+                status='co'))
+
+        sections = affiliation.section_set.prefetch_related(prefetch, 'experience_set__subtypes')
+
+        for section in sections:
+            section.completion_board_stuff()
+
+        context['sections'] = sections
+        context['requirements'] = sections[0].requirement_dict
```

Comment:
Should probably check that there is a section in sections to get away from `IndexError`'s